packages/commons/src/config/featureFlagsConfig.ts

```diff
@@ -53,11 +53,32 @@ export type FeatureFlagAgreementApprovalPolicyUpdateConfig = z.infer<
   typeof FeatureFlagAgreementApprovalPolicyUpdateConfig
 >;
 
+export const FeatureFlagAdminClientConfig = z
+  .object({
+    FEATURE_FLAG_ADMIN_CLIENT: z
+      .enum(["true", "false"])
+      .default("false")
+      .transform((value) => value === "true"),
+  })
+  .transform((c) => ({
+    featureFlagAdminClient: c.FEATURE_FLAG_ADMIN_CLIENT ?? false,
+  }));
+
+export type FeatureFlagAdminClientConfig = z.infer<
+  typeof FeatureFlagAdminClientConfig
+>;
+
 type FeatureFlags = FeatureFlagSignalhubWhitelistConfig &
   FeatureFlagAgreementApprovalPolicyUpdateConfig &
-  FeatureFlagSQLConfig;
+  FeatureFlagSQLConfig &
+  FeatureFlagAdminClientConfig;
 
-export type FeatureFlagKeys = keyof FeatureFlags & `featureFlag${string}`;
+export type FeatureFlagKeys = keyof Pick<
+  FeatureFlags,
+  {
+    [K in keyof FeatureFlags]: K extends `featureFlag${string}` ? K : never;
+  }[keyof FeatureFlags]
+>;
```

Comment:
This diff is still unneeded, I think, revert to how it was before, it was improved in one of the last commits of Sandro's PR

```suggestion
export type FeatureFlagKeys = keyof FeatureFlags & `featureFlag${string}`;
```